src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1623,27 +1625,14 @@ public Status do_GET_Document(Database _db, String docID, String _attachmentName
                 String revID = getQuery("rev");  // often null
                 RevisionInternal rev = null;
                 if (isLocalDoc) {
+                    // query only -> not required synchronized
```

Comment:
As [getLocalDocument()](https://github.com/couchbase/couchbase-lite-java-core/blob/a35361bc47ecb6c55849694deece22fa4aaf7c62/src/main/java/com/couchbase/lite/store/SQLiteStore.java#L1826) does not access shard instance and SQLiteDatabae is thread-safe as default, I believe this method is thread-safe.